scripts/avg_checkpoints.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+"""Script to average values of variables in a list of checkpoint files.
+
+Given a list of model checkpoints, it generates a new checkpoint wiht
+parameters which are an arithmetic average of them.
+
+This is based on a script from Tensor2Tensor:
+https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/avg_checkpoints.py
+"""
+
+import argparse
+import os
+
+import numpy as np
+import six
+from six.moves import zip    # pylint: disable=redefined-builtin
+import tensorflow as tf
+
+from neuralmonkey.logging import log
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("checkpoints", type=str, nargs="+",
+                        help="Space-separated list of checkpoints to average.")
+    parser.add_argument("output_path", type=str,
+                        help="Path to output the averaged checkpoint to.")
+    args = parser.parse_args()
+
+    non_existing_chckpoints = []
+    for ckpt in args.checkpoints:
+        if not os.path.exists("{}.index".format(ckpt)):
+            non_existing_chckpoints.append(ckpt)
+    if non_existing_chckpoints:
+        raise ValueError(
+            "Provided checkpoints do not exist: {}".format(
+                ", ".join(non_existing_chckpoints)))
+
+    # Read variables from all checkpoints and average them.
+    log("Get list of variables:")
+    var_list = tf.contrib.framework.list_variables(args.checkpoints[0])
+    var_values, var_dtypes = {}, {}
+    for (name, shape) in var_list:
+        if not name.startswith("global_step"):
+            var_values[name] = np.zeros(shape)
+    for checkpoint in args.checkpoints:
+        reader = tf.contrib.framework.load_checkpoint(checkpoint)
+        for name in var_values:
+            tensor = reader.get_tensor(name)
+            var_dtypes[name] = tensor.dtype
+            var_values[name] += tensor
+        log("Read from checkpoint {}".format(checkpoint))
```

Comment:
Tohle si měl říct než jsi to dělal, ze kterýho checkpointu čteš. Pak bych jen lognul, že hotovo, nebo bych nic nelogoval.